library/common/templates/lib/controller/_ports.tpl

```diff
@@ -25,13 +25,30 @@ Ports included by the controller.
         {{- end }}
       {{- end }}
     {{- end }}
+    {{/* append the ports for each appAdditionalService - TrueCharts */}}
+    {{- if and $.Values.appAdditionalServicesEnabled $.Values.appAdditionalServices -}}
+      {{- range $name, $_ := $.Values.appAdditionalServices }}
+        {{- if .enabled -}}
+          {{- if kindIs "string" $name -}}
+            {{- $_ := set .port "name" (default .port.name | default $name) -}}
+            {{- else -}}
+            {{- $_ := set .port "name" (required "Missing port.name" .port.name) -}}
+          {{- end -}}
+          {{- $ports = mustAppend $ports .port -}}
+          {{- range $_ := .additionalPorts -}}
+            {{/* append the additonalPorts for each additional service */}}
+            {{- $ports = mustAppend $ports . -}}
+          {{- end }}
+        {{- end }}
+      {{- end }}
+    {{- end }}
```

Comment:
/custom/_appServicePorts.tpl ?

Reply:
Sure, will do

Reply:
Great, will merge right away afterwards, thanks for the effort! :)

Reply:
this one gonna take a little more work (for me at least) - because adding to the existing list of ports -- that will need passed to the custom template to have ports added, then passed back to this template to be rendered.

I know it's possible, I just don't have a clear understanding of how that works, yet

Reply:
Good point, I'll leave it as is then :)